src/Uno.UI.Toolkit/Diagnostics/DiagnosticsOverlay.DiagnosticViewContext.cs

```diff
@@ -0,0 +1,162 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.UI.Dispatching;
+using Uno.Foundation.Logging;
+
+namespace Uno.Diagnostics.UI;
+
+public sealed partial class DiagnosticsOverlay
+{
+	private sealed class Context : IDiagnosticViewContext, IDisposable
```

Comment:
Could an Rx query do this?

Reply:
Rx `IScheduler`  you mean ? Yes (except it seems that the `SchedulePeriodic` has been removed).

However I'm not sure however it worth to add a dependency to rx just the for the `IScheduler` interface while anyway I'll want to abstract it as we could need to add more members on the `IDiagnosticViewContext` later.